src/_pytest/assertion/util.py

```diff
@@ -231,10 +212,62 @@ def _diff_text(left: str, right: str, verbose: int = 0) -> List[str]:
         explanation += ["Strings contain only whitespace, escaping them using repr()"]
     # "right" is the expected base against which we compare "left",
     # see https://github.com/pytest-dev/pytest/issues/3333
-    explanation += [
-        line.strip("\n")
-        for line in ndiff(right.splitlines(keepends), left.splitlines(keepends))
-    ]
+
+    stripped_left = "".join(left.splitlines())
+    stripped_right = "".join(right.splitlines())
+    s = SequenceMatcher(None, stripped_left, stripped_right)
+
+    nlines_left = left.count("\n")
+    nlines_right = right.count("\n")
+
+    if s.ratio() < 0.30 or max(nlines_left, nlines_right) < 5:
+        explanation += [
+            line.strip("\n")
+            for line in ndiff(right.splitlines(keepends), left.splitlines(keepends))
+        ]
+    else:
+
+        def _text_header(header: str, screen_width: int, margin: int = 10) -> List[str]:
+            hlength = len(header)
+            lines = [
+                "=" * int((screen_width - hlength - margin) / 2)
+                + header
+                + "=" * int((screen_width - hlength - margin) / 2)
+            ]
+            if screen_width % 2 != 0:
+                lines[-1] += "="
+
+            return lines
+
+        explanation += _text_header(" ACTUAL ", screen_width)
+        explanation += list(left.split("\n"))
+        explanation += _text_header(" EXPECTED ", screen_width)
+        explanation += list(right.split("\n"))
```

Comment:
IIUC `str.center` can take care of most of this. Also can you explain why the margin (10) is needed?

```suggestion
        explanation += [
            " ACTUAL ".center(screen_width - 10, "="),
            *left.split("\n"),
            " EXPECTED ".center(screen_width - 10, "="),
            *right.split("\n"),
        ]
```